matrix/Dcm.hpp

```diff
@@ -28,20 +34,68 @@ class Dcm : public Matrix<Type, 3, 3>
 
     typedef Matrix<Type, 3, 1> Vector3;
 
+    /**
+     * Standard constructor
+     *
+     * Initializes to identity
+     */
     Dcm() : Matrix<Type, 3, 3>()
     {
         (*this) = eye<Type, 3>();
     }
 
+    /**
+     * Constructor from array
+     *
+     * @param other array
+     */
     Dcm(const Type *data_) : Matrix<Type, 3, 3>(data_)
     {
     }
 
+    /**
+     * Copy constructor
+     *
+     * @param other Matrix33 to set dcm to
+     */
     Dcm(const Matrix<Type, 3, 3> &other) : Matrix<Type, 3, 3>(other)
     {
     }
 
+    /**
+     * Constructor from quaternion
+     *
+     * Instance is initialized from quaternion representing
+     * transformation from inertial frame to body frame.
+     *
+     * @param q quaternion to set dcm to
+     */
     Dcm(const Quaternion<Type> & q) {
+        set_from_quaternion(q);
+    }
+
+    /**
+     * Constructor from euler angles
+     *
+     * Instance is initialized from angle tripplet (3,2,1) representing
```

Comment:
@priseborough yes, intrinsic rotation about z, y and then x. Will document this better.